app/controllers/changes_controller.rb

```diff
@@ -23,6 +23,34 @@ def resend
     render json: @change.send_notifications
   end
 
+  def html
+    @change_id=params[:change_id]
```

Comment:
So instead of directly interacting with the params here, I would add these methods to the list using the `set_change` filter on line 2. That will populate the `@change` variable before each method is called.